python/ray/train/tests/test_trainer.py

```diff
@@ -1,1301 +0,0 @@
-    trainer.run(train_func, callbacks=[kill_callback])
-    # Run 1: iter=0, counter=1, Successful
-    # Run 2: iter=1, counter=2, Successful
-    # Run 3: None, counter=2, Unsuccessful, starts training from beginning.
-    # Run 4: iter=0, counter=3, Successful
-    # Run 5: iter=1, counter=4, Successful
-    assert kill_callback.counter == 4
-
-    def train_func():
-        return 1
-
-    # Make sure Trainer is usable even after failure handling.
-    trainer.run(train_func)
-
-
-def test_worker_kill_checkpoint(ray_start_2_cpus):
-    test_config = TestConfig()
-
-    def train_func():
-        checkpoint = train.load_checkpoint()
-        if checkpoint:
-            epoch = checkpoint["epoch"]
-        else:
-            epoch = 0
-        print("Epoch: ", epoch)
-        for i in range(epoch, 2):
-            train.report(loss=1, iter=i)
-            train.save_checkpoint(epoch=i + 1)
-
-    trainer = Trainer(test_config, num_workers=2)
-    trainer.start()
-    kill_callback = KillCallback(fail_on=0, trainer=trainer)
-
-    trainer.run(train_func, callbacks=[kill_callback])
-
-    # Run 1: epoch=0, counter=1, Successful
-    # *Checkpoint is saved.*
-    # *Worker is killed*
-    # *Getting checkpoint fails. Workers are restarted from beginning*
-    # Run 2: epoch=0, counter=2, Successful
-    # Run 3: epoch=1, counter=3, Successful
-    assert kill_callback.counter == 3
-    assert trainer.latest_checkpoint["epoch"] == 2
-
-    trainer.shutdown()
-    trainer.start()
-
-    kill_callback = KillCallback(fail_on=1, trainer=trainer)
-    trainer.run(train_func, callbacks=[kill_callback])
-    # Run 1: epoch=0, counter=1, Successful
-    # *Checkpoint saved*
-    # *Latest checkpoint updated, epoch=1
-    # Run 2: epoch=1, counter=2, Successful
-    # *Checkpoint saved*
-    # *Worker is killed*
-    # *Getting checkpoint fails. Workers are restarted from last checkpoint.*
-    # Run 3: epoch=1, counter=3, Successful.
-    assert kill_callback.counter == 3
-    assert trainer.latest_checkpoint["epoch"] == 2
-
-    def train_func():
-        return 1
-
-    # Make sure Trainer is usable even after failure handling.
-    trainer.run(train_func)
-
-
-def test_multiple_run(ray_start_2_cpus):
-    config = TestConfig()
-
-    def train_1():
-        return 1
-
-    trainer = Trainer(config, num_workers=2)
-    trainer.start()
-
-    output_1 = trainer.run(train_1)
-    assert output_1 == [1, 1]
-
-    def train_2():
-        return 2
-
-    output_2 = trainer.run(train_2)
-    assert output_2 == [2, 2]
-
-
-def check_dataset_output(num_data, num_epochs, data_all_epochs):
-    assert all(len(worker_data) == num_epochs for worker_data in data_all_epochs)
-    for i in range(num_epochs):
-        epoch_data = []
-        for worker_data in data_all_epochs:
-            epoch_data.extend(worker_data[i])
-        assert len(epoch_data) == num_data
-        assert set(epoch_data) == set(range(num_data))
-
-
-def test_dataset(ray_start_4_cpus):
```

Comment:
should we rewrite these dataset tests with DataParallelTrainer?

Reply:
It should not be necessary, we are already testing for that logic in other tests implicitly

Reply:
looks like these are already being tested in `test_dataset_config`